app/scripts/modules/core/src/artifact/expectedArtifact.service.ts

```diff
@@ -0,0 +1,45 @@
+import { copy, module } from 'angular';
+
+import { IArtifact, IExpectedArtifact, IPipeline, IStage } from 'core/domain';
+import { PipelineConfigService } from 'core';
+
+export class ExpectedArtifactService {
+  constructor(private pipelineConfigService: PipelineConfigService) {
+    'ngInject';
+  }
+
+  public summarizeExpectedArtifact(expected: IExpectedArtifact): string {
+    if (!expected) {
+      return '';
+    }
+
+    const artifact = copy(expected.matchArtifact);
+    return Object.keys(artifact)
+      .filter((k: keyof IArtifact) => artifact[k])
```

Comment:
I would be surprised if the compiler couldn't infer the type here. Or maybe `copy` returns `any` and you lose the type? I bet you can drop the typing at least after the first `filter`.

Reply:
It can't seem to - I can drop it in the second filter since we are comparing against a string which it doesn't seem to mind